kobo/apps/subsequences/integrations/google/base.py

```diff
@@ -0,0 +1,128 @@
+from abc import ABC, abstractmethod
+from concurrent.futures import TimeoutError
+from typing import Any, Tuple
+
+import constance
+from google.cloud import storage
+from google.api_core.operation import Operation
+from googleapiclient import discovery
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.cache import cache
+
+from kobo.apps.trackers.utils import update_nlp_counter
+from kpi.utils.log import logging
+from .utils import google_credentials_from_constance_config
+from ...models import SubmissionExtras
+from ...constants import GOOGLE_CACHE_TIMEOUT, make_async_cache_key
+from ...exceptions import SubsequenceTimeoutError
+
+REQUEST_TIMEOUT = 10  # seconds
```

Comment:
What about making a setting with this constant? It could be also a Constance setting.
What do you think?

Reply:
Sure, sounds good